src/lib/server/pos-sessions.ts

```diff
@@ -0,0 +1,276 @@
+import { collections } from './database';
+import { runtimeConfig } from './runtime-config';
+import type { PosSession, PosSessionUser } from '$lib/types/PosSession';
+import { ObjectId } from 'mongodb';
+import { error } from '@sveltejs/kit';
+import type { PaymentMethod } from './payment-methods';
+import type { Currency } from '$lib/types/Currency';
+
+export async function getCurrentPosSession(): Promise<PosSession | null> {
+	return await collections.posSessions.findOne({ status: 'active' }, { sort: { openedAt: -1 } });
+}
+
+export async function getLastClosedSession(): Promise<PosSession | null> {
+	return await collections.posSessions.findOne({ status: 'closed' }, { sort: { closedAt: -1 } });
+}
+
+export async function openPosSession(params: {
+	cashOpening: { amount: number; currency: Currency };
+	user: PosSessionUser;
+}): Promise<ObjectId> {
+	if (!runtimeConfig.posSessionsEnabled) {
+		throw error(400, 'POS sessions are not enabled');
+	}
+
+	const existingSession = await getCurrentPosSession();
+	if (existingSession) {
+		throw error(400, 'There is already an active POS session');
+	}
+
+	const lastSession = await getLastClosedSession();
+	const yesterdayClosing = lastSession?.cashClosing;
+
+	if (
+		yesterdayClosing &&
+		(yesterdayClosing.amount !== params.cashOpening.amount ||
+			yesterdayClosing.currency !== params.cashOpening.currency)
+	) {
+		console.warn(
+			`Opening amount mismatch! Expected: ${yesterdayClosing.amount} ${yesterdayClosing.currency}, Got: ${params.cashOpening.amount} ${params.cashOpening.currency}`
+		);
+	}
+
+	const session: PosSession = {
+		_id: new ObjectId(),
+		status: 'active',
+		openedAt: new Date(),
+		openedBy: params.user,
+		cashOpening: params.cashOpening,
+		dailyIncomes: [],
+		dailyOutcomes: [],
+		xTickets: [],
+		createdAt: new Date(),
+		updatedAt: new Date()
+	};
+
+	await collections.posSessions.insertOne(session);
+	return session._id;
+}
+
+export async function calculateDailyIncomes(
+	session: PosSession
+): Promise<Map<PaymentMethod, { amount: number; currency: Currency }>> {
+	const orders = await collections.orders
+		.find({
+			createdAt: { $gte: session.openedAt },
+			status: 'paid'
+		})
+		.toArray();
+
+	const paidPayments = orders.flatMap((order) => order.payments).filter((p) => p.status === 'paid');
+
+	const incomes = new Map<PaymentMethod, { amount: number; currency: Currency }>();
+
+	paidPayments.forEach((payment) => {
```

Comment:
If the feature is intended for POS use, then we should only include POS payments (not all of them).